lib/shanty_docker_plugin/image_resolution_fallback.rb

```diff
@@ -0,0 +1,22 @@
+require 'shanty_docker_plugin/error'
+require 'shanty/logger'
+
+module DockerPlugin
+  # Methods for falling back to a different tag when resolving images
+  module ImageResolutionFallback
+    extend Shanty::Logger
+
+    module_function
+
+    def pull!(image, fallback_tag)
+      image.pull!
+    rescue Error::PullError
+      logger.warn(
```

Comment:
Yep, let's leave it for now and add as appropriate, docker is a bit more stable these days